WcaOnRails/app/webpacker/components/Auth/Signup/index.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useRef, useState } from 'react';
+import {
+  Button, Card, CardHeader, Container, Form, FormField, Message,
+} from 'semantic-ui-react';
+import ReCAPTCHA from 'react-google-recaptcha';
+import { GOOGLE_RECAPTCHA_SITE_KEY } from '../../../lib/wca-data.js.erb';
+import I18n from '../../../lib/i18n';
+
+export default function Signup() {
+  // initial is a minor hack to prevent fomantic CSS messing up the UI.
+  const [initial, setInitial] = useState(true);
```

Comment:
I'm in favour of naming things after what they are, rather then how they're used, so I'd recommend `hasClickedSignUpButton` instead of `initial`. (Note that this is the opposite boolean.)

Reply:
Makes sense, changed.